Deno deploy action needs upgrade to latest Deno version

When running from the default Deno version installed by the `setup-deno` action, the deploy action fails with:

```
error: The source code is invalid, as it does not match the expected hash in the lock file.
  Specifier: https://esm.sh/preact@10.15.1/jsx-runtime
  Lock file: /home/runner/work/trg/trg/deno.lock
Error: Process completed with exit code 10.
```

as seen in

- https://github.com/kkga/ter/actions/runs/5376914498/jobs/9754708635#step:4:163

Pinning Deno in the workflow run to the latest available version, which was working locally, helped to mitigate the issue in my fork.

```diff
       - name: Setup Deno
-        uses: denoland/setup-deno@v1.1.0
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: "1.34.3"
```

Reference:

- https://github.com/almereyda/trg/blob/1a9346396915caf607b54b758c2e90125660eea8/.github/workflows/deno-deploy.yml#L21-L24

--- a/.github/workflows/deno-deploy.yml
+++ b/.github/workflows/deno-deploy.yml
@@ -19,7 +19,9 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Setup Deno
-        uses: denoland/setup-deno@v1.1.0
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: v1.x
 
       - name: Build site
         run: deno task build

